src/controllers/api/thoughts.js

```diff
@@ -0,0 +1,88 @@
+const { Thoughts } = require("../../models");
+
+const getAllThoughts = async (req, res) => {
+  try {
+    const thoughts = await Thoughts.find({});
+
+    return res.json({ success: true, data: thoughts });
+  } catch (error) {
+    console.log(`[ERROR]: Failed to get all thoughts | ${error.message}`);
+  }
+};
+
+const getThoughtById = async (req, res) => {
+  try {
+    const { id } = req.params;
+
+    const getThought = await Thoughts.findOne({ _id: id });
+
+    if (!getThought) {
+      return res.status(404).json({ success: false });
+    }
+    res.status(200).json({ success: true, getThought });
+  } catch (error) {
+    console.log(error);
+
+    return res.status(500).json({ success: false });
+  }
+};
+
+const createThought = async (req, res) => {
+  try {
+    const { thoughtText, userName } = req.body;
+
+    if (thoughtText && userName) {
```

Comment:
Although there is nothing wrong with doing it this way, it is best practice to use the `unhappy` paths in the if statements e.g.
```suggestion
    if (!thoughtText || !userName) {
      return res.status(400).json({
        success: false,
        error: `Please provide a valid thought and username`,
      });
      
   await Thoughts.create({ userName, thoughtText });
      return res.json({ success: true });
```

This removes the need to use an else block and allows the code to flow better.